File: Python3/프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    
    w,h = 0,0
    for tw,th in sizes:
        cw = min(tw,th)
        ch = max(tw,th)
        
        w = max(w,cw)
        h = max(h,ch)
    
    return w*h
    
    
    
    
    
    
    
    
    
    
    
    
    
    
#     w = 0
#     h = 0
    
#     for size in sizes:
#         [a,b] = size
#         if a > b:
#             w = max(w,a)
#             h = max(h,b)
#         else:
#             w = max(w,b)
#             h = max(h,a)
    
#     answer = w * h
        
        
        
#     return answer

File: C++/프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int w = 0;
    int h = 0;
    
    for (auto& size : sizes) {
        if (size[0] > size[1]) {
            w = max(w, size[0]);
            h = max(h, size[1]);
        } else {
            w = max(w, size[1]);
            h = max(h, size[0]);
        }
       
    }
    
    answer = w * h;
    return answer;
}